src/software/ai/hl/stp/play/dynamic_plays/support_tactic_scorers/support_tactic_candidate.hpp

```diff
@@ -0,0 +1,141 @@
+#pragma once
+
+#include <set>
+
+#include "software/ai/hl/stp/play/dynamic_plays/support_tactic_scorers/support_tactic_scorer.h"
+#include "software/ai/hl/stp/tactic/pass_receiver/pass_receiver_tactic.h"
+#include "software/ai/hl/stp/tactic/fake_pass_receiver/fake_pass_receiver_tactic.h"
+#include "software/ai/hl/stp/tactic/cherry_picker/cherry_picker_tactic.h"
+#include "software/ai/hl/stp/tactic/disrupter/disrupter_tactic.h"
+
+/**
+ * A SupportTacticCandidate is a "type wrapper" that makes a type of tactic
+ * eligible for scoring and assignment as a support tactic in a DynamicPlay.
+ *
+ * It acts as the visitee for a SupportTacticScorer, which operates under
+ * the visitor pattern.
+ */
+template<typename TSupportTactic>
+class SupportTacticCandidate 
+{
+    static_assert(std::is_base_of<Tactic, TSupportTactic>::value, 
+                  "TSupportTactic must derive from Tactic");
+
+   public:
+    explicit SupportTacticCandidate();
+
+    /**
+     * Accepts a SupportTacticScorer and calls the scorer's `score` function 
+     * on itself. The scorer's returned score is applied to this candidate.
+     *
+     * @param scorer a SupportTacticScorer to use to score this candidate
+     */
+    void score(SupportTacticScorer &scorer);
+
+    /**
+     * Removes all scores applied to this candidate
+     */
+    void clearScores();
+
+    /**
+     * Gets the total score for this candidate.
+     * 
+     * The total score X(x_1, x_2, ..., x_n) summarizes all the individual 
+     * scores x_i that have been applied to this candidate. It is calculated 
+     * as a generalized mean of the form 
+     *
+     *                                  n
+     * X(x_1, x_2, ..., x_n) = ((1/n) * Σ sign(x_i) * |x_i|^(1/p))^p
+     *                                 i=1
+     *
+     * where the parameter p is > 0. As p gets larger, scores close to 0
+     * and negative scores will influence the total score more than values 
+     * far away from 0. 
+     *
+     * This parameter p is useful to us since negative scores are meant
+     * to penalize and scores close to 0 indicate low viability, so they
+     * should disproportionately impact the total score more so than any 
+     * single large positive score.
+     * 
+     * @return the total score for this support tactic candidate
+     */
+    double getTotalScore() const;
+
+    /**
+     * Returns a shared pointer to a newly constructed instance of the
+     * candidate's support tactic type
+     *
+     * @return a shared pointer to a newly constructed instance of the
+     * candidate's support tactic type
+     */
+    std::shared_ptr<TSupportTactic> createSupportTactic() const;
+
+   private:
+    std::vector<int> scores_;
+
+    // Parameter p in total score calculation
+    static constexpr double SINGLE_SCORE_INFLUENCE = 2.5;
```

Comment:
in the future, we might have this as a dynamic parameter and scale it up according to the risk value